src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -26,43 +30,136 @@ public class ConsoleLoggerProvider : ILoggerProvider, ISupportExternalScope
         /// </summary>
         /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
         public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
+            : this(options, Enumerable.Empty<ConsoleFormatter>()) { }
+
+        /// <summary>
+        /// Creates an instance of <see cref="ConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
+        /// <param name="formatters">Log formatters added for <see cref="ConsoleLogger"/> insteaces.</param>
+        public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnumerable<ConsoleFormatter> formatters)
         {
             _options = options;
             _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
+            SetFormatters(formatters);
 
             ReloadLoggerOptions(options.CurrentValue);
             _optionsReloadToken = _options.OnChange(ReloadLoggerOptions);
 
             _messageQueue = new ConsoleLoggerProcessor();
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (DoesConsoleSupportAnsi())
             {
-                _messageQueue.Console = new WindowsLogConsole();
-                _messageQueue.ErrorConsole = new WindowsLogConsole(stdErr: true);
+                _messageQueue.Console = new AnsiLogConsole(new AnsiSystemConsole());
```

Comment:
Do we need this double indirection? `new AnsiLogConsole(new AnsiSystemConsole());` ? Can we just fold that into a single class, just like we have for `AnsiParsingLogConsole`?